File: intro-prog/lista 6/ex02.py
"""
2. Escreva uma função que recebe uma lista vet e imprime apenas os valores pares.
"""

def evenList(vet):
    for elem in vet:
        if elem % 2 == 0 and elem > 0:
            print(f"{elem} ")

def main():
    evenList([2, 3, 5, 4, 10, 0])

main()

File: Algoritmos/Lista 2/ex02.c
/*2. Escreva uma função que recebe um inteiro positivo e imprime seus fatores primos (e quantidades).
void print_factors(int a)*/

#include <stdio.h>

void printFactors(int num){
    int i = 2;
    int cont = 0;

    while (num > 1){
        if(num % i == 0){
            cont ++;
            num = num/i;
        }
        else 
        {   if(cont > 0){
                printf("Quantidade de %d: %d\n", i, cont);
                cont = 0;
            }
            i++;
        }
    }

    printf("Quantidade de %d: %d\n", i, cont);
}

int main(){
    printFactors(132);
}